Endpoint for block/unblock administrators

Hrushon opened the pull request:
Endpoint has been created for the ability to block/unblock of the administrators.

https://www.notion.so/API-3c491e4e49a448f8b7cec978fbae13f6

Hrushon committed:
feat(api, core): endpoint for blocking admin

Endpoint has been created for the ability to block/unblock of the administrators.

Hrushon committed:
feat(api, core): endpoint for blocking admin

Endpoint has been created for the ability to block/unblock of the administrators.

Ivan-Skvortsov commented on src/api/routers/administrator.py:
```diff
@@ -145,3 +149,23 @@ async def administrator_change_role(
+    ) -> AdministratorResponse:
+        """Заблокировать администратора."""
+        current_admin = await self.authentication_service.get_current_active_administrator(token.credentials)
+        if current_admin.role is not Administrator.Role.ADMINISTRATOR:
+            raise AdministratorBlockError
+        if current_admin.id == administrator_id:
+            raise AdministratorSelfBlockError
```
Давай вынесем это в сервис

Hrushon replied:
ок

Hrushon committed:
feat(api, core): endpoint for blocking admin

Endpoint has been created for the ability to block/unblock of the administrators.

Ivan-Skvortsov commented on src/core/services/administrator_service.py:
```diff
@@ -59,3 +59,17 @@ async def switch_administrator_role(self, changed_by: Administrator, administrat
 
         return await self.__administrator_repository.update(administrator.id, administrator)
+
+    async def block_administrator(self, blocks: Administrator, blocked_id: UUID) -> Administrator:
```
```suggestion
    async def block_administrator(self, blocked_by: Administrator, blocked_id: UUID) -> Administrator:

```

Hrushon committed:
fix(src): renamed the variable denoting the blocking admin

Co-authored-by: Ivan Skvortsov <pprofcheg@gmail.com>

Hrushon committed:
fix(src): renamed the variable denoting the blocking admin
